Fix test_get_trending_players mocked test player_id expectation

## Bug Description

The mocked test `test_get_trending_players` in `tests/test_sleeper_mcp_mocked.py` is failing because it expects a `player_id` field but receives `full_name` instead.

## Error Details

```
FAILED tests/test_sleeper_mcp_mocked.py::TestPlayerToolsMocked::test_get_trending_players - AssertionError: assert 'player_id' in {'full_name': 'Patrick Mahomes', 'position': 'QB', 'count': 150}
```

**Location**: `tests/test_sleeper_mcp_mocked.py::TestPlayerToolsMocked::test_get_trending_players`

## Root Cause

The test expects trending player data to contain a `player_id` field, but the mocked data includes `full_name` instead. This suggests a mismatch between:
- What the test expects from `get_trending_players`
- What the mock is returning

## Impact

- **Severity**: Low (test-only issue, doesn't affect production functionality)
- **Affects**: CI/CD pipeline test suite
- **Test Failure**: One of 3 remaining test failures (127/130 tests passing)

## Steps to Reproduce

1. Run: `uv run pytest tests/test_sleeper_mcp_mocked.py::TestPlayerToolsMocked::test_get_trending_players -xvs`
2. Test fails with assertion about missing `player_id` field

## Investigation Needed

1. Review actual structure returned by `get_trending_players`
2. Check if the mock data structure matches real API responses
3. Compare with the VCR-based integration test (which passes)
4. Determine correct player data structure for trending players

## Suggested Actions

1. Read the test to see what it's checking
2. Inspect the mock data setup for `get_trending_players`
3. Compare mock structure with real API responses (VCR cassettes)
4. Update either the mock data or test expectations to match reality
5. Ensure consistency across all player data structures

## Related

- Discovered during work on #90 (Phase 1 refactoring)
- Related to #98 (similar player field structure issue)
- The VCR-based integration test passes, suggesting the function works correctly
- Issue is with mock data structure, not actual functionality

--- a/scratchpads/issue-99-test-get-trending-players-mocked.md
+++ b/scratchpads/issue-99-test-get-trending-players-mocked.md
@@ -1,0 +1,59 @@
+# Issue #99: Fix test_get_trending_players mocked test player_id expectation
+
+**Issue Link**: https://github.com/GregBaugues/sleeper-mcp/issues/99
+
+## Problem Analysis
+
+The mocked test `test_get_trending_players` in `tests/test_sleeper_mcp_mocked.py:301` is failing because it expects a `player_id` field in the result, but receives `full_name` instead.
+
+### Error
+```
+FAILED tests/test_sleeper_mcp_mocked.py::TestPlayerToolsMocked::test_get_trending_players - AssertionError: assert 'player_id' in {'full_name': 'Patrick Mahomes', 'position': 'QB', 'count': 150}
+```
+
+## Root Cause
+
+After analyzing the code in `sleeper_mcp.py`, the `get_trending_players` function works as follows:
+
+1. Calls Sleeper API to get trending players (returns `player_id` and `count`)
+2. Enriches the data using `get_players_from_cache()` which returns full player data
+3. For players found in cache, it copies the full player data and adds the `count` field
+4. **The cached player data does NOT include `player_id` by default** - it has fields like `full_name`, `position`, `team`, etc.
+
+### The Mock Setup Issue
+
+In the test (lines 318-322), the mock cache returns:
+```python
+mock_cache.return_value = {
+    "4046": {"full_name": "Patrick Mahomes", "position": "QB"},
+    "4034": {"full_name": "Davante Adams", "position": "WR"},
+}
+```
+
+The test expects `player_id` in the result (line 328), but the function copies the cached player data which only has `full_name` and `position`.
+
+## Solution
+
+The mock cache data should include `player_id` to match the actual cached player structure. Looking at how the cache works, cached player data should include all player fields including `player_id`.
+
+The fix is to update the mock data to include `player_id`:
+
+```python
+mock_cache.return_value = {
+    "4046": {"player_id": "4046", "full_name": "Patrick Mahomes", "position": "QB"},
+    "4034": {"player_id": "4034", "full_name": "Davante Adams", "position": "WR"},
+}
+```
+
+## Implementation Plan
+
+1. Update mock cache data in test to include `player_id` field
+2. Verify the fix by running the specific test
+3. Run full test suite to ensure no regressions
+4. Commit and create PR
+
+## Related Issues
+
+- Similar to #98 (player field structure issue)
+- Part of #90 Phase 1 refactoring work
+- VCR-based integration test passes (indicates function works correctly in production)

--- a/tests/test_sleeper_mcp_mocked.py
+++ b/tests/test_sleeper_mcp_mocked.py
@@ -317,8 +317,16 @@ class TestPlayerToolsMocked:
             # Mock the cache function
             with patch("sleeper_mcp.get_players_from_cache") as mock_cache:
                 mock_cache.return_value = {
-                    "4046": {"full_name": "Patrick Mahomes", "position": "QB"},
-                    "4034": {"full_name": "Davante Adams", "position": "WR"},
+                    "4046": {
+                        "player_id": "4046",
+                        "full_name": "Patrick Mahomes",
+                        "position": "QB",
+                    },
+                    "4034": {
+                        "player_id": "4034",
+                        "full_name": "Davante Adams",
+                        "position": "WR",
+                    },
                 }
 
                 result = await sleeper_mcp.get_trending_players.fn(type="add")

